src/main/java/org/sonarlint/intellij/finding/hotspot/SecurityHotspotsPresenter.kt

```diff
@@ -47,7 +65,53 @@ class SecurityHotspotsPresenter(private val project: Project) {
             }
     }
 
+    fun refreshSecurityHotspotsForOpenFilesAsync() {
+        ProgressManager.getInstance()
+            .run(object : Task.Backgroundable(project, "Refreshing security hotspots...", false, ALWAYS_BACKGROUND) {
+                override fun run(indicator: ProgressIndicator) {
+                    refreshSecurityHotspotsForOpenFiles()
+                }
+            })
+    }
+
     fun refreshSecurityHotspotsForOpenFiles() {
+        try {
+            project.getOpenFiles().forEach { refreshSecurityHotspotsFor(project, it) }
+        } catch (e: Exception) {
+            showBalloon(project, SECURITY_HOTSPOTS_REFRESH_ERROR_MESSAGE, RefreshSecurityHotspotsAction("Retry"))
+            SonarLintConsole.get(project).error(SECURITY_HOTSPOTS_REFRESH_ERROR_MESSAGE, e)
+        }
         presentSecurityHotspotsForOpenFiles()
     }
+
+    private fun refreshSecurityHotspotsFor(project: Project, file: VirtualFile) {
```

Comment:
@damien-urruty-sonarsource This is a draft - basically I did what was done on taints, but there are some refactoring possible with the `ServerSecurityHotspotUpdater` class